core/src/main/scala/org/bitcoins/core/api/wallet/CoinSelectionAlgo.scala

```diff
@@ -32,7 +34,13 @@ object CoinSelectionAlgo {
            AccumulateSmallestViable,
            StandardAccumulate)
 
-  def fromString(str: String): Option[CoinSelectionAlgo] = {
+  override def fromStringOpt(str: String): Option[CoinSelectionAlgo] = {
     all.find(state => str.toLowerCase() == state.toString.toLowerCase)
   }
+
+  override def fromString(string: String): CoinSelectionAlgo = {
+    val algoOpt = fromStringOpt(string)
+    algoOpt.getOrElse(
+      sys.error(s"Could not find coin selection algorithm=${string}"))
```

Comment:
why `sys.error` over `IllegalArgumentExeception` or `RuntimeExeception`?

Reply:
sys.error is a runtime exception, it's the more "Scala way" to do it. I think there is a marginal benefit for using `sys.error()` because other runtimes -- such as scalajs -- have an easier way to decouple from jvm specific feature (java.lang.RuntimeException) and replace this call with whatever javascript throws for exceptions. :man_shrugging: . I've been trying to use that more.

Here's the definition:

```scala
package object sys {
  /** Throw a new RuntimeException with the supplied message.
   *
   *  @return   Nothing.
   */
  def error(message: String): Nothing = throw new RuntimeException(message)
```